### DISCUSSION TITLE
Release notes start from a tag that doesn't have a release

### DISCUSSION THREAD

**Discussion by RES-TophWells** [community/community#8381] [Other Feature Feedback, Questions, & Ideas] [answered]
This is occurring on a private organisation, so I can't provide links - please contact me directly if you need more information.

On two repositories that I run, I'm trying out the automatically generated release notes. They look great, but the release notes for `v1.22.6` show only the commits that have been added since the tag `v1.22.5`. This is a problem, since the tag v1.22.5 wasn't actually published as a GitHub release - we created a build with that version number for the testers, and tagged it at that time, but we didn't send it out to the end-users. In fact the previous version for which a release was created was `v1.22.1`. This means that all the PRs merged in between `v1.22.1` and `v1.22.5` were actually released, but they don't appear in the release notes!

If this isn't a sensible tagging strategy, I'm happy to change it. Should we be creating GitHub pre-releases for the "missing" tags?

I've read the discussion at #5975, but I think this is a different problem. That discussion centers around Github picking a tag that's too early; in my case it's picking a tag that's too late. A few posts in that discussion mention that it's possible to set the version range from the API, which would suit my purposes fine. What parameter should I set to get this result?

**Comment by MylesBorins** [comment, marked as answer]
In your position I would not be incrementing the version of a release if it isn't a release that consumers can adopt. SemVer has a way [to label pre-releases](https://semver.org/#spec-item-9) and you can use the GitHub pre-release feature to roll those out.

If you don't want to change your current release process, which is completely reasonable. You can use the [generate release notes api](https://docs.github.com/en/rest/reference/repos#generate-release-notes-content-for-a-release) directly to generate notes for any desired range

**Reply by RES-TophWells** [reply]
Thanks Myles! I discussed this with my team and we decided to change the release process to only tag released versions, as you suggest. I didn't consider using GitHub pre-releases as well; I'll look into it.

Thanks for the link to the API! I thought it would be part of the [create a release](https://docs.github.com/en/rest/reference/repos#create-a-release) API so I missed that release notes endpoint.

**Comment by jasonkarns** [comment]
I think this is a bigger problem than just the use case mentioned.

I am running into this exact problem, and it is caused when following the workflow that GitHub themselves advocate. The scenario is for a repository that houses GitHub Actions or Reusable Workflows.  Because GHA (and workflows) have no means of actually supporting semver properly, they approximate it by using tags. The recommended flow is to release a tag, say `v1.2.3` and to _also_ tag that same commit with `v1`. Then when `v1.2.4` comes along, they propose to force-push the `v1` tag to `v1.2.4`.

To end users, this gives the appearance of having semver with GitHub Actions and workflows. It allows the user to `uses: org/action@v1` instead of pinning to a specific patch version, commit sha or branch. However, this (again, _recommended_) workflow now causes the exact same problem as the OP. When releasing `v1.2.4` it generates release notes with a "compare" link of `v1...v1.2.4`. Which will be out of date the second the release is published. And then will be completely _wrong_, when v1.2.5 comes along and makes the compare link backwards!

Since the vN tag is a lightweight-only tag that is intended by design to be force-pushed, and will never have a Release associated, GitHub could solve the problem of their own making by making the generated release notes smarter in this regard.
